### ISSUE TITLE
ProveCommitSector must only accept its valid worker as a caller

### ISSUE THREAD

**Issue by nicola** [filecoin-project/specs-actors#527]
> ProveCommitSector must only accept its valid worker as a caller.

From code here:
https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/miner/miner_actor.go#L425-L427

I seem to understand that we don't check that the caller is the worker - only the worker should be able to send a ProveCommit

**Comment by nicola** [comment]
This is currently done for outsourcing, we need to check that a user can not create trouble for the storage miner.

**Comment by anorth** [comment]
This was changed to allow all in https://github.com/filecoin-project/specs-actors/pull/199 (ref https://github.com/filecoin-project/specs-actors/issues/189)

**Comment by anorth** [comment]
@nicola I would like to close this unless you have a concrete flaw/attack in mind.

**Comment by nicola** [comment]
This is a low priority to review on our end. We can close this for now, I will bring it up if it ever becomes an issue.

**Comment by nicola** [comment]
Reopening this one. We won't be able to analyse this on time. This should come as an improvement rather than the status quo.

Possible issues: someone that is doing selfish mining can do a "replay" attack of a PoRep by done by someone else, either by just reposting the proof or by fully regenerating their power.

**Comment by whyrusleeping** [comment]
@nicola I don't think thats worth being worried about. There are many other things that have to fall into place for the 'replaying' to have an impact (including the precommit).

Let's close this

**Comment by nicola** [comment]
Not that many things need to fall into place. In fact, an attacker can replay all the ProveCommits done in the past PreCommitChallengeDelay blocks.

Example of the attack:
- Consider PreCommitChallengeDelay to be the delay between PreCommit and ProveCommit
- At any point epoch, a miner can replay in their fork all the ProveCommits for which their fork has PreCommits that happened in the last PreCommitChallenge in other forks.

Actual example
- PreCommitChallengeDelay = 200 epochs
- A selfish mining attack is ~50 blocks
- The selfish miner can include in the 50 blocks all the ProveCommits that happened in the last 50 blocks (or more generally, all the ProveCommits in the last 200 blocks that would be considered valid for this fork)

**Comment by nicola** [comment]
If we use the ChainReference also for PoRep, this is mitigated (however, it will go against @whyrusleeping's proposal of delagating proof submission).

Resolutions:
- The way to go around this to solve my concerns and @whyrusleeping's delegation concerns is to postpone power at first PoSt.
- Add ChainReference to PoRep (but lose the ability to delegate PoRep submissions)

**Comment by nicola** [comment]
We are proposing to postpone power at first window post. Proposal is coming up soon.

**Comment by anorth** [comment]
Our issue for that is #414

**Comment by anorth** [comment]
Our issue for deferring power to first window post was #549, but we closed it with #648 because the original motivation for it (avoiding >24hr before first post) was addressed directly.

We should re-open that issue if confirmed that deferring power is the solution we want to take. @whyrusleeping @nicola

**Comment by nicola** [comment]
As per request of @jbenet, we decided that at least for now we should introduce chain reference for porep and apply this change.

He also proposed we do deferral of power #549, but I think it has a different priority.

**Comment by anorth** [comment]
Unnecessary after #884
